src/compiler/parser.ts

```diff
@@ -6728,6 +6794,26 @@ namespace ts {
                     return currentToken = scanner.scanJSDocToken();
                 }
 
+                function parseJSDocEntityName(createIfMissing = false): EntityName {

```

Comment:
This could use:

```ts
function createQualifiedName(entity: EntityName, name: Identifier): QualifiedName {
    const node = <QualifiedName>createNode(SyntaxKind.QualifiedName, entity.pos);
    node.left = entity;
    node.right = name;
    return finishNode(node);
}
```

And you can re-use `createQualifiedName` in `parseEntityName`. (`parseJSDocEntityName` should probably be near `parseEntityName` since they are similar.)